Makes lubed floors have the wet overlay, take two

PKPenguin321 opened the pull request:
Remake of #12767 but without the commit fuckyness (hopefully).

Relevant thread: https://tgstation13.org/phpBB/viewtopic.php?f=9&t=5071

Reasoning:
Makes lube not invisible so you can at least be aware of where it is on the floor and try to avoid it.
HOWEVER, you can't tell it apart from water, meaning if you try to avoid it by walking over it, you slip anyways. The slip is still as strong as ever.
This also opens the door to some mindgames by allowing you to spray water on the floor, then warning people that it's lube. :honk:

PKPenguin321 committed:
Makes lubed floors have the wet overlay, take two

Remake of #12767 but without the commit fuckyness. See #12767 for details. That's #12767.

MrStonedOne commented:
How about this:

Make lube have less stun time. but make it stack with water so that lube+water = old stun time.

Then you can choose between an invisible slip, or a long slip.

ExcessiveUseOfCobblestone commented:
but slips are really lame in general.

PKPenguin321 commented:
feedback should go in the thread that i linked in the OP
https://tgstation13.org/phpBB/viewtopic.php?f=9&t=5071
https://tgstation13.org/phpBB/viewtopic.php?f=9&t=5071
https://tgstation13.org/phpBB/viewtopic.php?f=9&t=5071
this thread to be precise

MrStonedOne commented:
How about this:

Make lube have less stun time. but make it stack with water so that lube+water = old stun time.

Then you can choose between an invisible slip, or a long slip.

Lumipharon commented:
invisible slips are gay. Everyone agrees, even the people that use it.
This is a simple fix that means you aren't as likely to get ggnore'd by something you have no way to know is there.

Shit like dermal patch got removed for the same reason, and atleast with that you could still see it with examine.

Metacide commented:
Glad someone is doing this. And hardly a huge nerf as it's still a long slip, and as Steelpoint points out:

>Making it visable akin to water will still make it powerful, as people will attempt to walk over the 'water' and in fact end up slipping themselves right into the comforting embrace of a energy sword.

Robustin commented:
Any nerf to lube is fine by me. As neat as it is killing deathsquads with a chemical that takes 5 seconds to make, shits been stupidly strong for too long.

bgobandit commented:
:-1:

xxalpha commented on code/game/turfs/simulated.dm:
```diff
@@ -17,7 +17,7 @@
 		return
 	wet = wet_setting
-	if(wet_setting == TURF_WET_WATER)
+	if(wet_setting == TURF_WET_WATER || TURF_WET_LUBE)
```
This is not how it works. Do this instead:
```
(wet_setting == TURF_WET_WATER) || (wet_setting == TURF_WET_LUBE)
```

PKPenguin321 replied:
Just curious, what exactly is the benefit of changing it to either of these?
The code should work the way I've done it already.

duncathan replied:
no, no it won't

duncathan replied:
`if(wet_setting == TURF_WET_WATER || TURF_WET_LUBE)`
this checking two separate conditions; they are:
`wet_setting == TURF_WET_WATER` and `TURF_WET_LUBE`
`if(TURF_WET_LUBE)` will evaluate to true as long as it's a truthy value (anything other than 0, null, or "")

PKPenguin321 replied:
but it literally does work
i just compiled it and tested it ingame
???

i guess i could fix it to what you guys are suggesting anyways since those should work too, but i don't really see a point and i REALLY don't want to delve back into git hell again :cry:

xxalpha replied:
It works because it always validates, but the logic is wrong.

PKPenguin321 committed:
fixes the logic

thanks xxalpha
i'd describe the change more in-depth but it's literally one line, hopefully you can figure out what i changed

Amemelius commented:
Actually a buff cloaked as a nerf. The sole drawback to using lube is not knowing when it's gone. Now you can trick people into thinking it's water on top of knowing when and where not to slip.

Huge buff that will result in more lube nerfs down the line.

phil235 commented:
:+1:

PKPenguin321 commented:
so uh
@Razharas ?

Cheridan commented:
@PKPenguin321 no, that's the reason

anyway, i think https://tgstation13.org/phpBB/viewtopic.php?f=9&t=5071#p130990 should be the prevailing logic here. things should be visible so that its possible to react to them in meaningful ways

PKPenguin321 commented:
nice
